connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala

```diff
@@ -194,6 +200,9 @@ private[sql] class AvroDeserializer(
       case (FLOAT, FloatType) => (updater, ordinal, value) =>
         updater.setFloat(ordinal, value.asInstanceOf[Float])
 
+      case (FLOAT, DoubleType) => (updater, ordinal, value) =>
+        updater.setDouble(ordinal, value.asInstanceOf[Float].toString.toDouble)
```

Comment:
@cloud-fan  I made a change here. 

When it comes to converting maximum or minimum values ​​from float to double, if using the implicit toDouble directly, the result will not meet expectations. For example, the difference in the result is actually a very large number.

![image](https://github.com/user-attachments/assets/ee42baf4-a6cf-4dee-9579-6b5382c8bc6f)

And I think this problem is also faced in the processing of parquet.
https://github.com/apache/spark/blob/d431d4034219f2c84c105e1894eebe03745bf105/sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRowConverter.scala#L330

Reply:
At the same time, I think the logic of converting float to double(for float, `case x: NumericType` is matched, and then `toDouble` is called directly) in our cast expression is also worth discussing the implementation, which is similar to this problem.

https://github.com/apache/spark/blob/6d32472b2a4225bd847953a679d3be2b077d212e/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/Cast.scala#L1008-L1031

Reply:
what's the behavior of SQL CAST? We should be consistent with that

Reply:
It calls toDouble directly, just like in the code above `numeric.toDouble(b)`.

<img width="611" alt="image" src="https://github.com/user-attachments/assets/44272b0a-2366-46fb-a39d-26342dc5a230">

Reply:
But the actual difference will be very large.

Reply:
Well, I agree to be consistent with Expression Cast to avoid unnecessary trouble. Update after soon.  

(But actually if we want to ensure accurate calculations, we'd better avoid using toDouble directly.)